### PR TITLE
WP-91 allow editing & deleting only for co-editors and media owners

### PR DESCRIPTION
**Pull request by muli**

### CODE REVIEW

File: lib/KalturaModel.php [hunk 1]
```diff
@@ -286,4 +286,24 @@ protected function getMaxPager() {
...
 	}
+
+	public function canLoggedInUserEditMedia($entry) {
+		if($this->isMediaOwner($entry)) {
+			return true;
+		}
+
+		$loggedInUserId = strtolower(KalturaHelpers::getLoggedUserId());
+		$entryCoPublishers = array_map('strtolower', explode(',', $entry->entitledUsersEdit));
+		if(in_array($loggedInUserId, $entryCoPublishers)) {
```
Comment (rkreich) [inline lib/KalturaModel.php]: Better add the 3rd parameter for `in_array` to have a strict comparison

File: view/library/browse.php [hunk 2]
```diff
@@ -75,8 +78,11 @@
 					<?php
 					$sendToEditorUrl = KalturaHelpers::generateTabUrl( array( 'tab' => 'kaltura_browse', 'kaction' => 'sendtoeditor', 'entryIds' => array( $mediaEntry->id ) ) );
 					?>
-
-					<div id="entryId_<?php echo esc_attr( $mediaEntry->id ); ?>" class="showName" title="<?php echo esc_attr( 'Click to edit' ); ?>">
+					<?php
+					$canUserEditMedia = $kalturaModel->canLoggedInUserEditMedia($mediaEntry);
```
Comment (rkreich) [inline view/library/browse.php]: You should init `$canUserEditMedia` in the controller and not in the view.
`$kalturaModel` should not be referenced here.

File: css/admin.css [hunk 3]
```diff
@@ -139,6 +139,14 @@ ul#kaltura-browse li .showName {
 	margin:      0 5px 0 5px;
 }
 
+ul#kaltura-browse li .entryTitle {
```
Comment (rkreich) [inline css/admin.css]: There is duplication in CSS here, maybe use `.entryTitle` (or change it to something else) as the base class (that will always be added to the div) and `.showName` to initialize `kalturaEditableName` plugin.

File: view/library/browse.php (hunk 2 above)
Comment (muli) [inline view/library/browse.php]: Please take a look and let me know if this solution is OK.
I am not sure whether it's smart to add properties to the entry object, but I couldn't think of a simpler solution...

Comment (rkreich) [inline view/library/browse.php]: I can't think of something better without over-complicating it.

**Comment by muli** [comment]
@rkreich - fixed according to your comments.
Let me know if it's OK.

**Comment by rkreich** [comment]
Yep, you can merge

**Pull request merged**